### PR TITLE
Parser: support storage engine and VALUE IN MAXVALUE

### PR DESCRIPTION
**Pull request by yangwenmai**
### What problem does this PR solve? 

fix issues: #500 

MySQL Syntax:
`ALTER TABLE ..., ADD PARTITION ( PARTITION ... STORAGE ENGINE text_string ...)`

### Bad Case

```sql
ALTER TABLE d_n.t_n WITHOUT VALIDATION , ADD PARTITION ( PARTITION ident VALUES LESS THAN ( MAXVALUE ) STORAGE ENGINE text_string MAX_ROWS 12 );
ALTER TABLE d_n.t_n WITH VALIDATION , ADD PARTITION NO_WRITE_TO_BINLOG (PARTITION ident VALUES LESS THAN MAXVALUE STORAGE ENGINE = text_string, PARTITION ident VALUES LESS THAN ( MAXVALUE ) (SUBPARTITION text_string MIN_ROWS 11));
ALTER TABLE d_n.t_n WITHOUT VALIDATION , ADD PARTITION ( PARTITION ident VALUES IN ( MAXVALUE ) STORAGE ENGINE text_string MAX_ROWS 12 );
ALTER TABLE d_n.t_n WITH VALIDATION , ADD PARTITION NO_WRITE_TO_BINLOG ( PARTITION ident VALUES IN ( MAXVALUE ) STORAGE ENGINE text_string MAX_ROWS 12 );
ALTER TABLE d_n.t_n WITH VALIDATION , ADD PARTITION NO_WRITE_TO_BINLOG (PARTITION ident VALUES LESS THAN MAXVALUE STORAGE ENGINE = text_string, PARTITION ident VALUES IN ( MAXVALUE ) (SUBPARTITION text_string MIN_ROWS 11));
```

### Check List

Tests

- Unit test

### CODE REVIEW

File: parser.y [hunk 1]
```diff
@@ -3186,6 +3186,10 @@ PartDefOption:
 	{
 		$$ = &ast.TableOption{Tp: ast.TableOptionEngine, StrValue: $3.(string)}
 	}
+|   "STORAGE" "ENGINE" EqOpt StringName
+    {
+        $$ = &ast.TableOption{Tp: ast.TableOptionEngine, StrValue: $4.(string)}
+    }
```
Comment (kennytm) [inline parser.y]: Reindent these 4 lines with tabs.

File: parser.y [hunk 2]
```diff
@@ -3186,6 +3186,10 @@ PartDefOption:
 	{
 		$$ = &ast.TableOption{Tp: ast.TableOptionEngine, StrValue: $3.(string)}
 	}
+|   "STORAGE" "ENGINE" EqOpt StringName
```
Comment (leoppro) [inline parser.y]:
```suggestion
|	"STORAGE" "ENGINE" EqOpt StringName
```

File: parser.y [hunk 3]
```diff
@@ -3186,6 +3186,10 @@ PartDefOption:
...
+    {
```
Comment (leoppro) [inline parser.y]:
```suggestion
	{
```

File: parser.y [hunk 4]
```diff
@@ -3186,6 +3186,10 @@ PartDefOption:
...
+        $$ = &ast.TableOption{Tp: ast.TableOptionEngine, StrValue: $4.(string)}
```
Comment (leoppro) [inline parser.y]:
```suggestion
	$$ = &ast.TableOption{Tp: ast.TableOptionEngine, StrValue: $4.(string)}
```

File: parser.y (hunk 1 above)
Comment (leoppro) [inline parser.y]:
```suggestion
	}
```

File: parser.y (hunk 2 above)
Comment (leoppro) [reply parser.y]: i thought i have permission to push, but i was wrong

**Pull request merged**